pyccel/parser/extend_tree.py

```diff
@@ -0,0 +1,245 @@
+"""Extended AST with CommentLine nodes
+======================================
+
+"""
+
+from numpy import array, logical_and, where
+from ast   import AST, If as IfNode, parse
+from pyccel.errors.errors import Errors
+from pyccel.errors.messages import INVALID_PYTHON_SYNTAX
+
+class CommentLine(AST):
+    """"New AST node representing a comment line"""
+
+    _fields = ("s",)
+
+    def __init__(self, s, lineno, col_offset):
+        super().__init__()
+        self.s = s
+        self.lineno     = lineno
+        self.col_offset = col_offset
+
+class CommentMultiLine(CommentLine):
+    """"New AST node representing a multi-line comment"""
+
+
+def get_comments(code): 
```

Comment:
Can we have a proper docstring with `Parameters` and `Returns`?

Same comment for the other functions in this module...

Reply:
Anyway, we should merge this important PR now and leave minor improvements for later.